general/solomiia_tymoshchuk/Task8.java

```diff
@@ -0,0 +1,46 @@
+package general.solomiia_tymoshchuk;
+
+/**
+ * Write a program that prints all numbers from 1 to N, that are divisible by 2, 3 or by both.
+ * If a number is divisible by only 2 - print it and add in parentheses “(by 2)”,
+ * if only by 3 - “(by 3)”, if by both numbers - “(by 2 and 3)”.
+ * Every number should be printed on the new line.
+ * Create two versions of the program: one should use the for loop, second - the while loop.
+ **/
+public class Task8 extends Task7 {
+
+    void checkNumbers() {
+        for (int m : getNewarr()) {
```

Comment:
Such approach requires us to scan all numbers twice: first time we need to go over all numbers and add them to a list, second time - select numbers we are interested in. Also, it increases the memory usage since we need to store these values somewhere. It will be much simpler to directly use the `for (int i = 1; i <= getNumber(); i++) {` cycle here.